### PR TITLE
Use native promise

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,57 +1,15 @@
...
-            // descriptor) has been closed. Not all streams will emit this.
-            stream.on("close", resolve);
-
-            // When the end() method has been called, and all data has been flushed
-            // to the underlying system, this event is emitted.
-            stream.on("finish", resolve);
-        });
-    }));
+  const on = promisify(stream.on, stream);
+  return Promise.race(['error', 'end', 'close', 'finish'].map(on));
```
Comment (epeli) [inline index.js]: How does this handle the error case? I think es6-promisify assumes that the function passed to it is a error first callback, but these are not. They are events. The first argument can be anything.

Comment (graingert) [inline index.js]: https://nodejs.org/api/stream.html#stream_event_error
The 'error' event returns an Error object. All the other events call back with nothing.

Comment (epeli) [inline index.js]: Right. Bit too clever coding here for me :)

Comment (graingert) [inline index.js]: I can't see how else you'd use promisify on a stream

**Comment by epeli** [comment]
I all for migrating to native promises but there couple of issues with this PR

- It changes the API. The error is not wrapped anymore. Which means the stream is lost when the error happend. The `promisePipe` now takes an array instead of directly accepting the promises.
- Removes comments from the stream events
- Remove the justPromise helper
- es6-promisify is not added to package.json
  - I'd prefer this to not have any dependencies
- Have you ran the tests?

**Comment by graingert** [comment]
I think I prefer this api. It matches Promise.all and Promise.race

**Comment by epeli** [comment]
Dude, easier way would be just to replace `Q` with the global `Promise` from the original source now and avoid the code golfing :)

**Comment by graingert** [comment]
@epeli no deps, and tests fixed.

File: index.js [hunk 2]
```diff
@@ -1,57 +1,43 @@
...
-            stream.on("end", resolve);
+function streamPromise(stream) {
+  if (stream === process.stdout || stream === process.stderr) {
+    return Promise.resolve(stream);
+  }
+
+  function on(evt) {
+    function executor(resolve, reject) {
+      const fn = evt === 'error' ?
+        err => reject(err) :
```
Comment (epeli) [inline index.js]: Why not directly reject here with the stream error? Eg. `reject(new StreamError(err, stream))` no need catch and rethrow later.

**Comment by epeli** [comment]
Looking good!

**Comment by epeli** [comment]
Released as 2.0.0 to npm.

**Pull request merged**